### PR TITLE
use new HttpContent each time for HttpRequestMock

### PR DESCRIPTION
**Pull request by xieofxie**
## Description

use new HttpContent each time: when used by some custom action, they may dispose it so we can't reuse HttpContent

## Specific Changes

add HttpResponseMockContent to generate HttpContent

## Testing

update TestScriptTests_HttpRequestMock.test.dialog for repeating bytearray response

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/HttpResponseMockContent.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using static Microsoft.Bot.Builder.Dialogs.Adaptive.Testing.HttpRequestMocks.HttpResponseMock;
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/HttpResponseMockContent.cs:6]: I don't remember what "Using Static" does. Can you add a comment here. 

Consider not using this. The 4 C# programmers doing this code review on the conference call - none of us know what this does.

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/HttpResponseMockContent.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Testing.HttpRequestMocks
+{
+    public class HttpResponseMockContent
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/HttpResponseMockContent.cs:10]: Public classes need /// Comments

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/HttpResponseMockContent.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                    break;
+                case ContentTypes.ByteArray:
+                    content = Convert.FromBase64String(httpResponseMock.Content == null ? string.Empty : httpResponseMock.Content.ToString());
+                    break;
+                default:
+                    throw new NotSupportedException($"{httpResponseMock.ContentType} is not supported yet!");
+            }
+        }
+
+        public HttpContent GetHttpContent()
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/HttpResponseMockContent.cs:38]: All public methods need /// comments

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/SequenceResponseManager.cs [hunk 4]
```diff
@@ -11,32 +11,23 @@ namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Testing.HttpRequestMocks
     public class SequenceResponseManager
     {
         private int _id;
-        private List<HttpContent> _contents;
+        private List<HttpResponseMockContent> _contents;
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/SequenceResponseManager.cs:14]: Initialize this here as an emptyp list.

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/HttpResponseMockContent.cs (hunk 1 above)
Comment (xieofxie) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/HttpResponseMockContent.cs]: Removed.

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/SequenceResponseManager.cs [hunk 5]
```diff
@@ -8,39 +8,37 @@
...
+    /// <summary>
+    /// Manage Sequence Response for HttpRequestSequenceMock.
+    /// </summary>
     public class SequenceResponseManager
     {
         private int _id;
-        private List<HttpContent> _contents;
+        private List<HttpResponseMockContent> _contents = new List<HttpResponseMockContent>();
 
         public SequenceResponseManager(List<HttpResponseMock> responses)
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/SequenceResponseManager.cs:19]: >SequenceResponseManager [](start = 15, length = 23)

comments?

Comment (xieofxie) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/HttpRequestMocks/SequenceResponseManager.cs:24]: Added.

**Comment by mrivera-ms** [comment]
@xieofxie, please fix the conflicts so we can move forward with reviewing and merging.

**Comment by xieofxie** [comment]
@mrivera-ms Thanks! Updated.

**Pull request merged**